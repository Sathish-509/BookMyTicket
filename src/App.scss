
@import './styles/all.scss';

.App {
    text-align: center;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
  }

  .center {
    margin: auto;
    width: 50%;
    padding: 10px;
}

h1 {
  text-align: center;
}

.container {
  width: 600px;
  margin: auto;
}

.grid td {
  width: 80px;
  height: 50px;
  border: 2px solid orange;
  text-align: center;
  float: left;
  margin: 5px;
}

.grid td:hover {
  opacity: .7;
  cursor: pointer;
}

.reserved {
  background: orange;
  color: #fff;
  display: inline-block;
}

.reserved::after {
  content: 'RESERVED'
}

.available::after {
  content: 'AVAILABLE'
}

.available {
  color: #333;
}

li {
  list-style: none;
}
