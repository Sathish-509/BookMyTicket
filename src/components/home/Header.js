import React from 'react';
import { Modal, Button, FormControl } from 'react-bootstrap';
import * as bookingActions from './../../actions/bookingActions';
import * as eventActions from './../../actions/eventActions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DateTimePicker from 'react-datetime-picker';
import SearchBox from 'react-search-box';

class Header extends React.Component {
  constructor() {
    super();
    this.state = {
      show: false,
      chosenDetails: {},
      searchData: []
    };
    this.handleHide = this.handleHide.bind(this);
    this.saveEvent = this.saveEvent.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleHide() {
    this.setState({ show: false });
  }

  handleChange = selection => {
    debugger;
    selection ? console.log(selection.full_name) : console.log('reverted');
  };

  setValue(id, value) {
    debugger;
    let { chosenDetails } = this.state;
    chosenDetails.eventdate = new Date();
    switch (id) {
      case 'EVENTNAME':
        chosenDetails.eventname = value;
        break;
      case 'EVENTLOCATION':
        chosenDetails.eventlocation = value;
        break;
      case 'EVENTDATE':
        chosenDetails.eventdate = value;
        break;
    }
    this.setState({ chosenDetails });
  }

  saveEvent() {
    debugger;
    this.setState({ chosenDetails: {}, show: false });
    this.props.actions.createEvent(this.state.chosenDetails);
  }

  render() {
    debugger;
    let searchData = this.props.listOfEvents || [];
    return (
      <div>
        <div className="header">
          <a href="#default" className="logo">
            BookMyTicket
          </a>
          <SearchBox
            data={searchData}
            onChange={this.handleChange}
            placeholder="Search for a string..."
            class="search-class"
            searchKey="eventname"
            width={300}
            height={40}
          />
          <div className="header-right">
            <a href="/">Home</a>
            <a href="/ticketdetails">Ticket Details</a>
            <Button
              bsStyle="primary"
              bsSize="sm"
              onClick={() => this.setState({ show: true })}
              style={{ marginTop: '10px' }}
            >
              Create Event
            </Button>
          </div>
        </div>
        <div className="modal-container">
          <Modal
            show={this.state.show}
            onHide={this.handleHide}
            container={this}
            aria-labelledby="contained-modal-title"
          >
            <Modal.Header closeButton>
              <Modal.Title id="contained-modal-title">
                Enter Event Information
              </Modal.Title>
            </Modal.Header>
            <Modal.Body>
              <div className="offset-sm-3 field-set">
                <div className="row">
                  <div className="col-sm-6">
                    <div className="form-group">
                      <label htmlFor="eventname">Event Name: </label>
                      <FormControl
                        type="text"
                        value={this.state.chosenDetails.eventname || ''}
                        onChange={e =>
                          this.setValue('EVENTNAME', e.currentTarget.value)
                        }
                      />
                    </div>
                  </div>
                </div>
                <div className="row">
                  <div className="col-sm-6">
                    <div className="form-group">
                      <label htmlFor="eventlocation">Event Location: </label>
                      <FormControl
                        type="text"
                        value={this.state.chosenDetails.eventlocation || ''}
                        onChange={e =>
                          this.setValue('EVENTLOCATION', e.currentTarget.value)
                        }
                      />
                    </div>
                  </div>
                </div>
                <div className="row">
                  <div className="col-sm-6">
                    <div className="form-group">
                      <label htmlFor="eventdate">Event Date: </label>
                      <DateTimePicker
                        onChange={date => this.setValue('EVENTDATE', date)}
                        value={this.state.chosenDetails.eventdate || new Date()}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </Modal.Body>
            <Modal.Footer>
              <Button onClick={this.handleHide}>Close</Button>
              <Button bsStyle="primary" onClick={this.saveEvent}>
                Save Event
              </Button>
            </Modal.Footer>
          </Modal>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state, ownProps) {
  debugger;
  return {
    listOfEvents: state.event,
    bookings: state.booking
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      { ...bookingActions, ...eventActions },
      dispatch
    )
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
