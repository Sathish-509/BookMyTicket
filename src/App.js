import React, { Component } from 'react';
import './App.scss';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import BookingDetailsContainer from './containers/BookingDetailsContainer';
import HomePageContainer from './containers/HomePageContainer';
import TicketDetailsContainer from './containers/TicketDetailsContainer';
import Header from './components/home/Header';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <div>
          <BrowserRouter>
            <Switch>
              <Route exact path="/" component={HomePageContainer} />
              <Route
                path="/bookingdetails"
                component={BookingDetailsContainer}
              />
              <Route path="/ticketdetails" component={TicketDetailsContainer} />
            </Switch>
          </BrowserRouter>
        </div>
      </div>
    );
  }
}

export default App;
