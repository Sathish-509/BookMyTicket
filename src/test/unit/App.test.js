import React from 'react';
import { shallow } from 'enzyme';
import App from './../../App';
import chai, { expect } from 'chai';
import chaiEnzyme from 'chai-enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { shallowToJson } from 'enzyme-to-json';
import Header from './../../components/home/Header';

it('should render two div', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('div')).to.have.lengthOf(2);
  expect(wrapper.find('div').length).to.equal(2);
});

it('should render one div with the correct class applied', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('.App')).to.have.lengthOf(1);
});

/*
it('renders correctly', () => {
  const wrapper = shallow(<App />);   
expect(shallowToJson(wrapper)).toMatchSnapshot();
});*/
