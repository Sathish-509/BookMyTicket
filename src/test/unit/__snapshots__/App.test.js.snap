// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <div>
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/bookingdetails"
        />
        <Route
          component={[Function]}
          path="/ticketdetails"
        />
      </Switch>
    </BrowserRouter>
  </div>
</div>
`;
